dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -260,6 +235,27 @@ def apply(self, sdfg):
             outputs[e.src_conn] = e
             output_set[e.data.data] = e.src_conn
 
+        # Access nodes that need to be reshaped
+        reshapes: Set(str) = set()
+        for aname, array in nsdfg.arrays.items():
+            if array.transient:
+                continue
+            edge = None
+            if aname in inputs:
+                edge = inputs[aname]
+                if len(array.shape) > len(edge.data.subset):
```

Comment:
Why the reshape is happening when `len(array.shape) > len(edge.data.subset)` and not the opposite as well?

Reply:
In this case, there is definitely no unsqueeze possible. For the other case (whether it is an unsqueeze or reshape), the edge is kept for the _check_strides test below